core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -98,30 +109,59 @@ private boolean validatePrincipalLogin() {
 
   private boolean validateHdfsSettingParity(Map<String, String> optionsMap) {
     String serverHadoopConfDirPath;
-    if (optionsMap.containsKey(ValidateEnv.HADOOP_CONF_DIR_OPTION.getOpt())) {
-      serverHadoopConfDirPath = optionsMap.get(ValidateEnv.HADOOP_CONF_DIR_OPTION.getOpt());
+    if (optionsMap.containsKey(HADOOP_CONF_DIR_OPTION.getOpt())) {
+      serverHadoopConfDirPath = optionsMap.get(HADOOP_CONF_DIR_OPTION.getOpt());
     } else {
       serverHadoopConfDirPath = System.getenv(HADOOP_CONF_DIR_ENV_VAR);
     }
     if (serverHadoopConfDirPath == null) {
-      System.out.println("Path to server-side hadoop configuration unspecified,"
-          + " aborting validation for HDFS properties.");
-      return false;
+      System.err.println("Path to server-side hadoop configuration unspecified,"
+          + " skipping validation for HDFS properties.");
+      return true;
     }
     String serverCoreSiteFilePath = serverHadoopConfDirPath + "/core-site.xml";
     String serverHdfsSiteFilePath = serverHadoopConfDirPath + "/hdfs-site.xml";
 
     // If Configuration does not contain the key, then a {@link RuntimeException} will be thrown
     // before calling the {@link String#split} method.
-    String[] clientHaoopConfFilePaths = Configuration.get(PropertyKey.UNDERFS_HDFS_CONFIGURATION).split(":");
+    String[] clientHaoopConfFilePaths =
+        Configuration.get(PropertyKey.UNDERFS_HDFS_CONFIGURATION).split(":");
     String clientCoreSiteFilePath = clientHaoopConfFilePaths[0];
     String clientHdfsSiteFilePath = clientHaoopConfFilePaths[1];
     if (clientCoreSiteFilePath == null || clientCoreSiteFilePath.isEmpty()
         || clientHdfsSiteFilePath == null || clientHdfsSiteFilePath.isEmpty()) {
       System.out.println("Cannot locate the client-side hadoop configurations,"
-          + " aborting validation for HDFS properties.");
+          + " skipping validation for HDFS properties.");
+      return true;
+    }
+    return compareConfigurations(clientCoreSiteFilePath, serverCoreSiteFilePath)
+        && compareConfigurations(clientHdfsSiteFilePath, serverHdfsSiteFilePath);
+  }
+
+  private boolean compareConfigurations(String clientConfigFilePath, String serverConfigFilePath) {
+    ConfigurationFileParser parser = new ConfigurationFileParser();
+    Map<String, String> serverCoreSiteProps = parser.parseXmlConfiguration(serverConfigFilePath);
+    Map<String, String> clientCoreSiteProps = parser.parseXmlConfiguration(clientConfigFilePath);
+    boolean matches = true;
+    for (Map.Entry<String, String> prop : clientCoreSiteProps.entrySet()) {
```

Comment:
You can use [MapDifference](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Maps.html#difference-java.util.Map-java.util.Map-) for the comparison